code/modules/modular_computers/file_system/programs/pai/radio.dm

```diff
@@ -37,11 +37,15 @@
 	VUEUI_SET_CHECK(data["listening"], host.radio.broadcasting, ., data)
 	VUEUI_SET_CHECK(data["frequency"], format_frequency(host.radio.frequency), ., data)
 
-	LAZYINITLIST(data["channels"])
+	var/list/pai_channels = list()
 	for(var/ch_name in host.radio.channels)
-		var/ch_stat = host.radio.channels[ch_name]
-		VUEUI_SET_CHECK(data["channels"][ch_name], !!(ch_stat & host.radio.FREQ_LISTENING), ., data)
-	
+		var/list/channel_info = list(
+			"name" = ch_name,
+			"listening" = (host.radio.channels[ch_name] & host.radio.FREQ_LISTENING)
+		)
+		pai_channels[++pai_channels.len] = channel_info
+	VUEUI_SET_CHECK(data["channels"], pai_channels, ., data)
```

Comment:
`VUEUI_SET_CHECK` does not work for lists. You must use `VUEUI_SET_CHECK_LIST` instead.